tests/test_tags.py

```diff
@@ -206,6 +206,22 @@ def test_close_or_remove_client(hlwm):
     assert int(hlwm.get_attr('tags.focus.frame_count')) == 1
 
 
+def test_floating_focused_change(hlwm):
+    # create a floating client which is not focused
+    # because the tiling layer is focused
+    hlwm.call('rule floating=on focus=off')
+    floating_winid, _ = hlwm.create_client()
+    assert hlwm.get_attr('tags.focus.floating_focused') == hlwm.bool(False)
+    assert 'focus' not in hlwm.list_children('clients')
+
+    # switch to floating layer
+    hlwm.call('attr tags.focus.floating_focused on')
```

Comment:
Extend the test to see what happenes when we set it to "off" again maybe? Would feel more complete.
Also: What happens if we have two floating clients? How gets focus?

Reply:
I'll add the test cases later.

Reply:
I've added a testcase and I'm  also disallowing the floating layer to be focused if there are no floating windows at all.

Reply:
Awesome thank you for extending the tests!
Denying the floating layer focusing broke another test case though

Reply:
I've fixed it (the test case was the automatic generation and execution of commands via the completions)

Reply:
Thanks!